models/solution/route.rb

```diff
@@ -55,6 +55,34 @@ def insert_stop(vrp, stop, index, idle_time = 0)
         shift_route_times(idle_time + stop.activity.duration, index)
       end
 
+      def remove_stop(vrp, stop, index)
+        removed_stop = stops.delete_at(index)
+        if index != 0 && index != stops.size
```

Comment:
Why don't we need to do any corrections if the deleted stop is the first or the last?

I think we need to at least correct, total_time and total_travel_time. Actually, we need to do correct every stop.info.current_time/distance etc as well.

In fact, we can use the same code with minor modifications. if the index is 0, then there is no previous_stop (i.e., `nil`) and when previous_stop is nil then `new_travel_time` is 0 and previous_end is removed_stop.info.begin_time (i.e., `previous_end = previous_stop&.info.end_time || removed_stop.info.begin_time`). A similar modification is needed for previous_matrix_index.

Similarly if removed_stop is the last stop, then we need to correct the totals, but instead of doing them manually we can apply the above logic and this time base it on the existence of  `next_stop`/`next_matrix_index` etc.

Let me know what you think.